### PR TITLE
NPM version URLs

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/938)

### CODE REVIEW

File: analyzer/src/funTest/assets/projects/synthetic/npm-version-urls-expected-output.yml [hunk 1]
```diff
@@ -0,0 +1,4843 @@
+---
```
Comment (fviernau) [inline analyzer/src/funTest/assets/projects/synthetic/npm-version-urls-expected-output.yml]: I see some shortcomings with testing this level of detail (parsing of different kind of URLs) on the abstraction level of package manager like redundancy, readability, execution speed. What do you think about the following propasal?

Make a compromise between this approach and a unit test by testing
just the parsing of a single 'package.json' which does contain only the minimum needed 
for illustrating the behavior (under test). In particular skipping the installation of the modules and testing 
the code of NPM.kt line 173 - 253 (ideally excluding also the OKHttpRequest)

Comment (sschuberth) [reply analyzer/src/funTest/assets/projects/synthetic/npm-version-urls-expected-output.yml]: That basically makes absolute sense, but I'd say it's out of scope of this change (which is based on the existing `NpmTest`) and should rather be tackled in the context of issue https://github.com/heremaps/oss-review-toolkit/issues/755.

Comment (fviernau) [reply analyzer/src/funTest/assets/projects/synthetic/npm-version-urls-expected-output.yml]: sure, makes sense to do it separately later on.

**Comment by sschuberth** [comment]
Merging despite the bogus Travis CI failure, as [the build actually succeeded](https://travis-ci.com/heremaps/oss-review-toolkit/builds/87398298#L2383).

**Pull request merged**